### PR TITLE
Adding ovirt_event Module (#51760)

### PR DESCRIPTION
**Pull request by nasx**
##### SUMMARY

Submitting module ovirt_event to create and delete events in oVirt/RHV (original request is #51760)

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

ovirt_event

##### ADDITIONAL INFORMATION

This PR was split from #51761.

### CODE REVIEW

**Comment by machacekondra** [comment]
shipit

**Review by mwperina — APPROVED** [review]
shipit

**Comment by machacekondra** [comment]
@Akasurde @gundalow Can you please merge?

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: (c) 2019, Ansible Project
+#
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py]: Replace line 5 to 20 with 

```
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
from __future__ import absolute_import, division, print_function
__metaclass__ = type
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: ovirt_event
+short_description: Create or delete an event in oVirt/RHV
+author: "Chris Keller (@nasx)"
+version_added: "2.8"
+description:
+    - "Create or delete an event in oVirt/RHV."
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py]:
```suggestion
    - "This module can be used to create or delete an event in oVirt/RHV."
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+options:
+    state:
+        description:
+            - "Should the event be present/absent."
+            - "The wait option must be set to false when state is absent."
+        choices: ['present', 'absent']
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py:26]:
```suggestion
        choices: ['present', 'absent']
        type: str
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        default: present
+
+    description:
+        description:
+            - "Message for the event."
+            - "Required when state is present."
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py:33]:
```suggestion
            - "Required when state is present."
         type: str
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+    severity:
+        description:
+            - "Severity of the event."
+            - "Required when state is present."
+        choices: ['error', 'normal', 'warning']
+        default: normal
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py:41]: Include type for other parameters as well.

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+        description:
+            - "The id of the VM associated with this event."
+extends_documentation_fragment: ovirt
+'''
+
+EXAMPLES = '''
+# Examples don't contain the auth parameter for simplicity,
+# look at the ovirt_auth module to see how to reuse authentication.
+
+# Create an event
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py]:
```suggestion
- name: Create an event
  ovirt_event:
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+- ovirt_event:
+    state: present
+    description: "The file system /home on host xyz is almost full!"
+    origin: "mymonitor"
+    custom_id: 123456789
+    severity: warning
+
+# Create an event and link it to a specific object
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py]:
```suggestion
- name: Create an event and link it to a specific object
  ovirt_event:
```

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+short_description: Create or delete an event in oVirt/RHV
+author: "Chris Keller (@nasx)"
+version_added: "2.8"
+description:
+    - "Create or delete an event in oVirt/RHV."
+options:
+    state:
+        description:
+            - "Should the event be present/absent."
+            - "The wait option must be set to false when state is absent."
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/ovirt/ovirt_event.py]:
```suggestion
            - "The C(wait) option must be set to false when state is absent."
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Minor changes. Rest LGTM

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py (hunk 1 above)
Comment (nasx) [reply lib/ansible/modules/cloud/ovirt/ovirt_event.py]: Resolved w/ commit 4fbd13e

File: lib/ansible/modules/cloud/ovirt/ovirt_event.py (hunk 5 above)
Comment (nasx) [reply lib/ansible/modules/cloud/ovirt/ovirt_event.py:41]: Resolved w/ commit 4fbd13e

**Comment by machacekondra** [comment]
shipit

**Comment by machacekondra** [comment]
@Akasurde can you please re-review?

**Comment by mwperina** [comment]
shipit

**Comment by machacekondra** [comment]
@Akasurde @gundalow Can you please merge?

**Comment by machacekondra** [comment]
shipit

**Comment by machacekondra** [comment]
bot_status

**Comment by mwperina** [comment]
Abhijeet/John could you please merge?

@Akasurde @gundalow

**Review by Akasurde — APPROVED** [review]
shipit

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.8
Thank you all!

**Pull request merged**